libs/designer-ui/src/lib/arrayeditor/index.tsx

```diff
@@ -85,8 +85,9 @@ export const ArrayEditor: React.FC<ArrayEditorProps> = ({
   useEffect(() => {
     arrayType === ArrayType.COMPLEX
       ? initializeComplexArrayItems(initialValue, itemSchema, setItems, setIsValid, setCollapsed)
-      : initializeSimpleArrayItems(initialValue, setItems, setIsValid, setCollapsed);
-  }, [arrayType, initialValue, itemSchema]);
+      : initializeSimpleArrayItems(initialValue, itemSchema.type, setItems, setIsValid, setCollapsed);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
Why no dependencies?

Reply:
this was being called onBlur, so if the user typed an invalid value type in the expanded array, it would be converted to collapsedArray automatically. Instead it should now only do it on mount